src/main/java/io/github/jamsesso/jsonlogic/evaluator/expressions/DateArithmeticExpression.java

```diff
@@ -0,0 +1,62 @@
+package io.github.jamsesso.jsonlogic.evaluator.expressions;
+
+import io.github.jamsesso.jsonlogic.evaluator.JsonLogicEvaluationException;
+import io.github.jamsesso.jsonlogic.utils.BigDecimalOperations;
+import io.github.jamsesso.jsonlogic.utils.DateOperations;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+public class DateArithmeticExpression implements PreEvaluatedArgumentsExpression {
+  public static final DateArithmeticExpression DATE_ADD = new DateArithmeticExpression("+");
+  public static final DateArithmeticExpression DATE_SUBTRACT = new DateArithmeticExpression("-");
+
+  private final String key;
+
+  public DateArithmeticExpression(String key) {
+    this.key = key;
+  }
+
+  @Override
+  public String key() {
+    return key;
+  }
+
+  @Override
+  public Object evaluate(List arguments, Object data) throws JsonLogicEvaluationException {
+    if (arguments.isEmpty()) {
+      return null;
+    }
+
+    if (arguments.size() == 3) {
+      try {
+        OffsetDateTime date = DateOperations.fromString((String) arguments.get(0));
+        BigDecimal intervalToAdd = (BigDecimal) arguments.get(1);
+        ChronoUnit interval = ChronoUnit.valueOf((String) arguments.get(2));
```

Comment:
These 2 can be clubbed into a single argument of [ISO8601 Duration](https://en.wikipedia.org/wiki/ISO_8601#Durations)